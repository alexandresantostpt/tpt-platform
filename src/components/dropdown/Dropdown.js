import PropTypes from 'prop-types'
import React, { useState } from 'react'
import styled from 'styled-components'

import { not } from '@utils/functions'

import DropdownPicture from './picture/DropdownPicture'
import DropdownList from './list/DropdownList'
import Icon, { StyledIcon } from '@components/icon/Icon'
import If from '@components/if/If'

const StyledDropdown = styled.div`
    display: inline-block;
    position: relative;
    z-index: 1;
    &:hover {
        cursor: pointer;
    }

    ${StyledIcon} {
        color: #52575e;
        font-size: 2.5rem;
    }
`

const Dropdown = ({ children, icon, image, imageAlt }) => {
    const [expanded, updateExpanded] = useState(false)

    const toggleExpanded = () => updateExpanded(not(expanded))

    return (
        <StyledDropdown onMouseEnter={toggleExpanded} onMouseLeave={toggleExpanded}>
            <If condition={!!image.length} el={<Icon icon={icon} />}>
                <DropdownPicture image={image} imageAlt={imageAlt} />
            </If>
            <DropdownList expanded={expanded}>{children}</DropdownList>
        </StyledDropdown>
    )
}

Dropdown.defaultProps = {
    icon: '',
    image: '',
    imageAlt: ''
}

Dropdown.propTypes = {
    children: PropTypes.oneOfType([PropTypes.array, PropTypes.func, PropTypes.object, PropTypes.string]).isRequired,
    icon: PropTypes.string,
    image: PropTypes.string,
    imageAlt: PropTypes.string
}

export default Dropdown
