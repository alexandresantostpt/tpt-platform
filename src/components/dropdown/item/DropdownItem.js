import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'

const StyledDropdownItem = styled.li`
    border-top: 1px solid #dcdcdc;
    list-style: none;
    padding: ${({ bigger }) => (bigger ? '.75rem' : '.25rem')} 2rem;
    text-align: ${({ center }) => (center ? 'center' : 'left')};
    &:hover {
        background: rgba(0, 0, 0, 0.1);
    }
`

const DropdownItem = ({ onClick, ...props }) => <StyledDropdownItem onClick={onClick} {...props} />

DropdownItem.defaultProps = {
    bigger: false,
    center: false,
    onClick: null
}

DropdownItem.propTypes = {
    bigger: PropTypes.bool,
    center: PropTypes.bool,
    children: PropTypes.oneOfType([PropTypes.array, PropTypes.func, PropTypes.object, PropTypes.string]).isRequired,
    onClick: PropTypes.func
}

export default DropdownItem
