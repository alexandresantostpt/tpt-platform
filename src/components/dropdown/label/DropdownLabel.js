import PropTypes from 'prop-types'
import React from 'react'
import styled, { css } from 'styled-components'

const typeStyles = css`
    ${({ dark }) => {
        if (dark) {
            return `
                font-size: 1rem;
                font-weight: 500;
            `
        }
        return `
            font-size: .9rem;
            font-weight: 100;
        `
    }}
`

const StyledDropdownLabel = styled.span`
    color: #707070;
    display: block;
    ${typeStyles};
    width: 385px;
`

const DropdownLabel = ({ dark, children }) => <StyledDropdownLabel dark={dark}>{children}</StyledDropdownLabel>

DropdownLabel.defaultProps = {
    dark: false
}

DropdownLabel.propTypes = {
    children: PropTypes.string.isRequired,
    dark: PropTypes.bool
}

export default DropdownLabel
