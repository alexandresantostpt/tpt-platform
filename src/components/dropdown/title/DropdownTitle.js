import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'

const StyledDropdownTitle = styled.span`
    color: #707070;
    display: block;
    font-size: 0.875rem;
    font-weight: 300;
    opacity: 0.53;
    text-transform: uppercase;
`

const DropdownTitle = ({ children }) => <StyledDropdownTitle>{children}</StyledDropdownTitle>

DropdownTitle.propTypes = {
    children: PropTypes.string.isRequired
}

export default DropdownTitle
