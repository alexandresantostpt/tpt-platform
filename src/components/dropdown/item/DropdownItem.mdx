---
menu: Dropdown
name: DropdownItem
route: /dropdown/item
---
import { Playground, PropsTable } from 'docz'

import DropdownItem from './DropdownItem'

import '@cfg/scripts'
import '@cfg/style'

# DropdownItem
<PropsTable of={ DropdownItem }/>

## Basic usage

```
<DropdownItem>johnny.silva@teresaperez.com.br</DropdownItem>
```

<Playground>
    <DropdownItem>johnny.silva@teresaperez.com.br</DropdownItem>
</Playground>

### Bigger
Show item with more padding

```
<DropdownItem bigger>Johnny Silva</DropdownItem>
```

<Playground>
    <DropdownItem bigger>Johnny Silva</DropdownItem>
</Playground>
