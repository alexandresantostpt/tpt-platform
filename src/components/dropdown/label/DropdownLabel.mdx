---
menu: Dropdown
name: DropdownLabel
route: /dropdown/label
---
import { Playground, PropsTable } from 'docz'

import DropdownLabel from './DropdownLabel'

import '@cfg/scripts'
import '@cfg/style'

# DropdownLabel
<PropsTable of={ DropdownLabel }/>

## Basic usage

```
<DropdownLabel>johnny.silva@teresaperez.com.br</DropdownLabel>
```

<Playground>
    <DropdownLabel>johnny.silva@teresaperez.com.br</DropdownLabel>
</Playground>

### Bigger
Show item with a bigger font-size and font-weight

```
<DropdownLabel bigger>Johnny Silva</DropdownLabel>
```

<Playground>
    <DropdownLabel dark>Johnny Silva</DropdownLabel>
</Playground>
