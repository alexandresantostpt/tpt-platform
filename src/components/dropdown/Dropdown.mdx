---
menu: Dropdown
name: Dropdown
route: /dropdown
---
import { Playground, PropsTable } from 'docz'

import { i18n } from '@i18n'

import Dropdown from './Dropdown'
import DropdownItem from './item/DropdownItem'
import DropdownLabel from './label/DropdownLabel'
import DropdownTitle from './title/DropdownTitle'

import '@cfg/scripts'
import '@cfg/style'

# Dropdown
<PropsTable of={ Dropdown }/>

## Basic usage

```
<Dropdown image="IMAGE_URL" imageAlt="IMAGE_ALT">
    <DropdownItem>
        <DropdownTitle>{ i18n.t('titles.myProfile') }</DropdownTitle>
    </DropdownItem>
    <DropdownItem bigger>
        <DropdownLabel dark>Johnny Silva</DropdownLabel>
        <DropdownLabel>johnny.silva@teresaperez.com.br</DropdownLabel>
    </DropdownItem>
</Dropdown>
```

<Playground>
    <Dropdown image="https://api.adorable.io/avatars/100/tpt-platform" imageAlt="Johnny Silva">
        <DropdownItem>
            <DropdownTitle>{ i18n.t('titles.myProfile') }</DropdownTitle>
        </DropdownItem>
        <DropdownItem bigger>
            <DropdownLabel dark>Johnny Silva</DropdownLabel>
            <DropdownLabel>johnny.silva@teresaperez.com.br</DropdownLabel>
        </DropdownItem>
        <DropdownItem bigger>
            <DropdownLabel dark>{ i18n.t('labels.activitiesRegistry') }</DropdownLabel>
            <DropdownLabel>Atualizado Ã s 8:45</DropdownLabel>
        </DropdownItem>
    </Dropdown>
</Playground>

### Using icon
It's possible use icon instead image

<Playground>
</Playground>
