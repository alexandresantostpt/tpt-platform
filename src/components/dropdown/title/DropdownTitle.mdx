---
menu: Dropdown
name: DropdownTitle
route: /dropdown/title
---
import { Playground, PropsTable } from 'docz'

import DropdownTitle from './DropdownTitle'

import '@cfg/scripts'
import '@cfg/style'

# DropdownTitle
<PropsTable of={ DropdownTitle }/>

## Basic usage

```
<DropdownTitle>Meu perfil</DropdownTitle>
```

<Playground>
    <DropdownTitle>Meu perfil</DropdownTitle>
</Playground>
