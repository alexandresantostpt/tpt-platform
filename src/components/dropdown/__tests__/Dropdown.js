import React from 'react'
import renderer from 'react-test-renderer'

import { i18n } from '@i18n'

import Dropdown from '@components/dropdown/Dropdown'
import DropdownItem from '@components/dropdown/item/DropdownItem'
import DropdownLabel from '@components/dropdown/label/DropdownLabel'
import DropdownTitle from '@components/dropdown/title/DropdownTitle'

describe('Tests for Dropdown component', () => {
    it('Should rendering without crash ', () => {
        const tree = renderer
            .create(
                <Dropdown image="https://api.adorable.io/avatars/100/tpt-platform" imageAlt="Johnny Silva">
                    <DropdownItem>
                        <DropdownTitle>{i18n.t('titles.myProfile')}</DropdownTitle>
                    </DropdownItem>
                    <DropdownItem bigger>
                        <DropdownLabel dark>Johnny Silva</DropdownLabel>
                        <DropdownLabel>johnny.silva@teresaperez.com.br</DropdownLabel>
                    </DropdownItem>
                    <DropdownItem bigger>
                        <DropdownLabel dark>{i18n.t('labels.activitiesRegistry')}</DropdownLabel>
                        <DropdownLabel>Atualizado Ã s 8:45</DropdownLabel>
                    </DropdownItem>
                </Dropdown>
            )
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})
