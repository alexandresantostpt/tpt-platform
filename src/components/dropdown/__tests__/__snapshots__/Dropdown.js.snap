// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Dropdown component Should rendering without crash  1`] = `
<div
  className="Dropdown__StyledDropdown-sc-14kwj8x-0 iZGQRX"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <img
    alt="Johnny Silva"
    className="DropdownPicture__StyledDropdownPicture-ylb9qy-0 iHpJFt"
    src="https://api.adorable.io/avatars/100/tpt-platform"
  />
  <ul
    className="DropdownList__StyledDropdownList-vjd4uo-0 iDpmEs"
  >
    <li
      className="DropdownItem__StyledDropdownItem-sc-102yxuy-0 jhIdjc"
      onClick={null}
    >
      <span
        className="DropdownTitle__StyledDropdownTitle-y9o5oh-0 hFdNKe"
      >
        Meu perfil
      </span>
    </li>
    <li
      className="DropdownItem__StyledDropdownItem-sc-102yxuy-0 kBLyiT"
      onClick={null}
    >
      <span
        className="DropdownLabel__StyledDropdownLabel-scn6ga-0 fJmcGN"
      >
        Johnny Silva
      </span>
      <span
        className="DropdownLabel__StyledDropdownLabel-scn6ga-0 gxUkFn"
      >
        johnny.silva@teresaperez.com.br
      </span>
    </li>
    <li
      className="DropdownItem__StyledDropdownItem-sc-102yxuy-0 kBLyiT"
      onClick={null}
    >
      <span
        className="DropdownLabel__StyledDropdownLabel-scn6ga-0 fJmcGN"
      >
        Registro de atividades
      </span>
      <span
        className="DropdownLabel__StyledDropdownLabel-scn6ga-0 gxUkFn"
      >
        Atualizado Ã s 8:45
      </span>
    </li>
  </ul>
</div>
`;
